### PR TITLE
pc - add stryker for mutation coverage checking

### PR DESCRIPTION
**Pull request by pconrad**
In this PR, we add Stryker for mutation testing of JavaScript code.

To use:
* `cd frontend`
* `npx stryker run`
* Open the report at the URL indicated.

### CODE REVIEW

File: frontend/stryker.conf.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    "reporters": [
+        "progress",
+        "clear-text",
+        "html"
+    ],
+    "coverageAnalysis": "off",
+    "jest": {
+        "projectType": "create-react-app"
+    },
+    "tempDirName": "stryker-tmp",
```
Comment (btk5h) [inline frontend/stryker.conf.json]: It looks like this is set to `.stryker-tmp` by default. ([docs](https://stryker-mutator.io/docs/stryker-js/configuration/#tempdirname-string)) Is there any reason this is being explicitly overridden?

If this is being intentionally overridden, the `.gitignore` should be updated to match.

**Review by btk5h — DISMISSED** [review]

File: frontend/stryker.conf.json (hunk 1 above)
Comment (pconrad) [reply frontend/stryker.conf.json]: Good catch.   I'll edit this out so we use the default, and change the .gitignore to match that.

Comment (pconrad) [reply frontend/stryker.conf.json]: Fixed

**Review by kphamm — APPROVED** [review]
LGTM

**Pull request merged**